src/types/message-store.ts

```diff
@@ -35,7 +35,13 @@ export interface MessageStore {
   /**
    * Queries the underlying store for messages that match the provided filter.
    */
-  query(tenant: string, filter: Filter, options?: MessageStoreOptions ): Promise<GenericMessage[]>;
+  query(
+    tenant: string,
+    filter: Filter,
+    dateSort?: DateSort,
```

Comment:
This is probably a point of contention, and I myself flip back & forth multiple times:

I am currently leaning towards a generic interface for the message store that minimizes the awareness of higher level constructs such as specific sorting enumerations. What do you think of:

```suggestion
    dateSort: { [key: string]: 1 | -1 },
```

This is purely intended as a point of discussion, the proposed change would:
1. Make the sort construct generic (though it could be harder to implement)
1. Follows design precedence of the filter being generic.
1. Follows design precedence of MongoDB (which filter also follows)
1. Make the parameter optional at this level, no specific reason, just my tendency of preferring required params if at all possible.

Reply:
I like this! Thanks for the suggestion, this is much more flexible... I'll look into implementing this.